### PR TITLE
[bridge] some refactoring for e2e tests

### PR DESCRIPTION
**Pull request by longbowlu**
## Description 
Some refactoring to make tests slightly faster:

1. consolidate `fn publish_coins_return_add_coins_on_sui_action` and `fn wait_for_server_to_be_up` in `crates/sui-bridge/src/e2e_tests/test_utils.rs` and `crates/sui-bridge/src/utils.rs`
2. refactor `publish_coins_return_add_coins_on_sui_action` to `publish_and_register_coins_return_add_coins_on_sui_action` which does token publish and token register.

## Test plan 

How did you test the new or updated feature?

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:

### CODE REVIEW

File: crates/sui-bridge/src/utils.rs [hunk 1]
```diff
@@ -125,20 +128,65 @@ pub async fn get_eth_signer_client(url: &str, private_key_hex: &str) -> anyhow::
...
+        .unwrap();
+
+    let senders = wallet_context.get_addresses();
+    // We want each sender to deal with one coin
+    assert!(senders.len() >= token_packages_dir.len());
+
+    // publish coin packages
+    let mut publish_tokens_tasks = vec![];
+
+    for (token_package_dir, sender) in token_packages_dir.iter().zip(senders.clone()) {
```
Comment (dariorussi) [inline crates/sui-bridge/src/utils.rs:156]: that zip is a bit strange for me, but all good.

**Review by dariorussi — APPROVED** [review]
LGTM

File: crates/sui-bridge/src/utils.rs (hunk 1 above)
Comment (longbowlu) [reply crates/sui-bridge/src/utils.rs:156]: landing now, if anything we will fix later

Comment (dariorussi) [reply crates/sui-bridge/src/utils.rs:156]: nothing to fix, I just though it was an unusual pattern and that is it. It was not worth to mention I guess.
I can elaborate but we can even talk about it just for fun

**Pull request merged**